src/recursive_shared_mutex.h

```diff
@@ -0,0 +1,104 @@
+#ifndef _RECURSIVE_SHARED_MUTEX_H
+#define _RECURSIVE_SHARED_MUTEX_H
+
+#include <cassert>
+#include <chrono>
+#include <condition_variable>
+#include <exception>
+#include <map>
+#include <mutex>
+#include <system_error>
+#include <thread>
+#include <tuple>
+#include <type_traits>
+
+
+/*
+ * This mutex has two levels of access, shared and exclusive. Multiple threads can own this mutex in shared mode but
```

Comment:
almost right for doxygen
At the top it should be

-/*
+/**

Reply:
fixed. thanks. i thought clang formatter would catch that for me. guess not